packages/core/src/tracing/idletransaction.ts

```diff
@@ -158,15 +158,24 @@ export class IdleTransaction extends Transaction {
             logger.log('[Tracing] cancelling span since transaction ended early', JSON.stringify(span, undefined, 2));
         }
 
-        const keepSpan = span.startTimestamp < endTimestamp;
-        if (!keepSpan) {
-          __DEBUG_BUILD__ &&
-            logger.log(
-              '[Tracing] discarding Span since it happened after Transaction was finished',
-              JSON.stringify(span, undefined, 2),
-            );
+        const spanStartedBeforeTransactionFinish = span.startTimestamp < endTimestamp;
+
+        const timeoutWithDelta = (this._finalTimeout + this._idleTimeout) / 1000;
+
+        const transactionDidNotExceedTimeout = endTimestamp - this.startTimestamp < timeoutWithDelta;
+        const spanDidNotExceedTimeout = span.endTimestamp - span.startTimestamp < timeoutWithDelta;
+        const spanEndedBeforeFinalTimeout = transactionDidNotExceedTimeout && spanDidNotExceedTimeout;
```

Comment:
I would expect to check if the span finished after the finalTimeout -> that implies the Tx with this span would end after finalTimeout -> we remove the span.
```js
       const spanEndedBeforeFinalTimeout = span.endTimestamp - this.startTimestamp < this._finalTimeout
```

Reply:
Yeah this is simpler, changing!